src/engine/strat_engine/blockdevmgr.rs

```diff
@@ -40,15 +42,45 @@ pub fn resolve_devices(paths: &[&Path]) -> io::Result<HashSet<Device>> {
         .collect()
 }
 
+#[derive(Debug)]
+pub struct BlkDevSegment {
+    pub blockdev: Rc<RefCell<BlockDev>>,
+    pub start: Sectors,
+    pub length: Sectors,
+}
+
+impl BlkDevSegment {
+    pub fn new(blockdev: Rc<RefCell<BlockDev>>, start: Sectors, length: Sectors) -> BlkDevSegment {
+        BlkDevSegment {
+            blockdev,
+            start,
+            length,
+        }
+    }
+
+    pub fn to_segment(&self) -> Segment {
+        Segment::new(*self.blockdev.borrow().device(), self.start, self.length)
+    }
+}
+
+/// Build a Vec<Segment> from BlkDevSegments. This is useful for calls
+/// to the devicemapper library.
+pub fn convert_to_dm_segs(bsegs: &[BlkDevSegment]) -> Vec<Segment> {
```

Comment:
Ah. It didn't click with me it referred to mathematical projection, I thought it was a noun -- i.e. it meant converting a type associated with this project to a dm type!

More options: `map_to_dm`, `map_to_dm_segs`, `to_dm_segs`, `make_dm_segs`, `build_vec_of_segments`.